azure-pipelines.yml

```diff
@@ -40,6 +40,11 @@ jobs:
     - bash: |
         ./build_tools/linting.sh
       displayName: Run linters
+    - bash: |
+        pip install ninja meson scipy
```

Comment:
Since https://github.com/scikit-learn/scikit-learn/pull/28721 we are checking the scipy version in `sklearn/meson.build` (the build dependencies versions in general), so you need scipy to run `meson setup`.

https://github.com/scikit-learn/scikit-learn/blob/4aeb191100f409c880d033683972ab9f47963fa4/sklearn/meson.build#L41-L45

I agree it is a bit surprising but I am not sure this is worth complicating the `meson.build`.